src/components/Modal/Modal.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useState } from "react";
+import ReactModal from "react-modal";
+
+import style from "./modal.module.scss";
+
+ReactModal.setAppElement("#root");
+
+const Modal = ({ modalProps, setModalProps, cities, setCities }) => {
```

Comment:
Too complicated implementation, You could add just isOpen, close(), and addCity(city) pops, index shouldn't be exposed to this component, it makes component highly coupled with its parent logic. Also it shouldn't edit the list of cities, just pass selected city to the parent's handler. Index also could be ommited since it is used only for cities editing